code/modules/vehicles/tonk.dm

```diff
@@ -0,0 +1,354 @@
+		return ..()
+	if(user in operators)
+		exit_tank(user)
+		return
+	var/position = alert("What seat would you like to enter?.",name,"pilot","gunner") //God I wish I had radials to work with
+	if(!position)
+		return
+	if(pilot && position == "pilot")
+		to_chat(user, "[pilot] is already controlling [src]!")
+		return
+	if(gunner && position == "gunner")
+		to_chat(user, "[gunner] is already controlling [src]!")
+		return
+	if(can_enter(user)) //OWO can they enter us????
+		to_chat(user, "You climb into [src] as a [position]!")
+		return enter(user, position) //Yeah i could do this with a define, but this way we're not using multiple things
+
+/obj/vehicle/tonk/proc/can_enter(var/mob/living/carbon/M) //NO BENOS ALLOWED
+	if(!M.IsAdvancedToolUser())
+		to_chat(M, "<span class='warning'>You don't have the dexterity to drive [src]!</span>")
+		return FALSE
+	if(!allowed(M))
+		to_chat(M, "<span class='warning'>Access denied.</span>")
+		return FALSE
+	var/obj/item/offhand = M.get_inactive_held_item()
+	if(offhand && !(offhand.flags_item & (NODROP|DELONDROP)))
+		to_chat(M, "<span class='warning'>You need your hands free to climb on [src].</span>")
+		return FALSE
+
+	if(!M.mind || !(!M.mind.cm_skills || M.mind.cm_skills.large_vehicle >= SKILL_LARGE_VEHICLE_TRAINED))
+		M.visible_message("<span class='notice'>[M] fumbles around figuring out how to get into the [src].</span>",
+		"<span class='notice'>You fumble around figuring out how to get into [src].</span>")
+	var/time = 10 SECONDS - 2 SECONDS * M.mind.cm_skills.large_vehicle
+	if(do_after(M, time, TRUE, 5, BUSY_ICON_BUILD) && M.Adjacent(src))
+		return TRUE
+
+/obj/vehicle/tonk/proc/enter(var/mob/user, var/position) //By this point, we've checked that the seats are actually empty, so we won't need to do that again HOPEFULLY
+	if(!user || !position)
+		return //Something fucked up. Whoops!
+	switch(position)
+		if("pilot")
+			user.forceMove(src)
+			pilot = user
+			operators += user
+		if("gunner")
+			user.forceMove(src)
+			gunner = user
+			operators += user
+
+/obj/vehicle/tonk/proc/exit_tank(var/mob/user) //By this point, we've checked that the seats are actually empty, so we won't need to do that again HOPEFULLY
+	if(!user)
+		return //Something fucked up. Whoops!
+	if(user == pilot)
+		user.forceMove(get_turf(src))
+		pilot = null
+		operators -= user
+	if(user == gunner)
+		user.forceMove(get_turf(src))
+		gunner = null
+		operators -= user
+
+/obj/vehicle/tonk/relaymove(mob/user, direction)
+	if(user.incapacitated() || user != pilot)
+		to_chat(user, "Youre not the pilot fucking idiot.")
+		return
+	if(world.time > l_move_time + move_delay)
+		. = step(src, direction)
+	update_icon()
+
+/obj/vehicle/tonk/Bump(atom/unluckyfuck, yes)
+	. = ..()
+	unluckyfuck.tank_act(src) //You poor motherfucker
+
+/obj/vehicle/tonk/Bumped(atom/unluckyfuck, yes)
+	return FALSE //Tonk BIG, THOG SAY NOT MOVE TONK WHEN BUMPED
+
+/client/MouseDown(object, location, control, params)
```

Comment:
Wouldn't it be better to have some kind of `vehicule` var in the mob that relay the click to any vehicule the mob is on instead of redefining these here ?
@spookydonut 
@DominikPanic 
@Rohesie

Reply:
That would be a bit saner, yes, tied to enter and leave procs to modify it.

Reply:
Yes, there' the click_intercept var for clients and mobs.

Reply:
I just did it that way for clarity

Reply:
This isn't helping clarity